### ISSUE TITLE
stack warnings on GHC 8.10.1: Failed to decode module interface

### ISSUE THREAD

**Issue by asr** [agda/agda#4571]
I'm getting many warnings like the following one when using `stack` and GHC 8.10.1:

```
$ stack install --stack-yaml=stack-8.10.1.yaml --system-ghc
...
Warning: Failed to decode module interface:
         /some-path/.stack-work/dist/x86_64-linux/Cabal-3.2.0.0/build/Agda/Auto/Auto.hi Decoding failure:
         Invalid magic: e49ceb0f
lid magic: e49ceb0f
...
```

It seems the warnings are related to https://github.com/commercialhaskell/stack/issues/5134.

**Comment by asr** [comment]
@L-TChen, can I assign you this issue?

**Comment by L-TChen** [comment]
Yes.

**Comment by andreasabel** [comment]
Is there anything we can do about this?  The problem seems to be higher up...

**Comment by L-TChen** [comment]
Given that it is an issue with stack, I am afraid that we are not able to solve it on our side. Either we ignore the failure or we revert to 8.8.3 ?

**Comment by L-TChen** [comment]
I would like to wait until a LTS for GHC 8.10.1 is released before doing anything. Maybe I am wrong, but it seems to me adding the Stackage support before LTS is available is rather premature.

**Comment by asr** [comment]
> Maybe I am wrong, but it seems to me adding the Stackage support before LTS is available is rather premature.

Currently Travis stages `compilation` and `main` use `stack`. We shouldn't wait for a LTS for a particular version of GHC for adding support for this version of GHC.

**Comment by andreasabel** [comment]
I suppose we just ignore these warnings until the issue is fixed higher up.  Do you have another plan of action in mind, @asr ?

**Comment by L-TChen** [comment]
I am building a GitHub workflow  using cabal’s nix-style local builds equivalent to the one on Travis. 

It usually takes a while for stack to catch the development of cabal, so it might be better to use cabal for the latest GHC support.

**Comment by asr** [comment]
> I suppose we just ignore these warnings until the issue is fixed higher up. Do you have another plan of action in mind, @asr ?

No.

There are different combinations for setting up our CI:

1. Travis + cabal-install
2. Travis + stack
3. GitHub workflow + cabal-install
4. GitHub workflow + stack

In relation to support the latest version released of GHC, the first one is the most straightforward but the other combinations also work.

**Comment by L-TChen** [comment]
> I am building a GitHub workflow using cabal’s nix-style local builds equivalent to the one on Travis.

~It is essentially blocked by #4216.~ (The `v1-` style might still be used, though.) The warnings emitted by stack are just ... warnings, and test suites are still working. I gave up moving the rest of scripts on Travis to GitHub workflow.

But, as the by-product, I realized the GitHub lifts the space restriction on cache, so our scripts now run faster on GitHub workflow. 

If the problem mentioned here persists for, say, few more weeks, then I will come back to this.

**Comment by L-TChen** [comment]
I don't see warnings any more on GHC 8.10.2 and stack 2.5.1 ([log](https://github.com/agda/agda/actions?query=workflow%3A%22Stack+build%22)). Laziness sometimes works.

**Comment by andreasabel** [comment]
I am observing warnings like this (in all kinds of projects) after I upgraded stack from 2.5.1 to 2.5.1.1:
```
$ stack --version
Version 2.5.1.1 x86_64 hpack-0.33.0
```
The upstream issue is here: https://github.com/commercialhaskell/stack/issues/5134
